plugin/pkg/admission/namespace/autoprovision/admission.go

```diff
@@ -83,6 +83,11 @@ func (p *Provision) Admit(a admission.Attributes) error {
 		Status: api.NamespaceStatus{},
 	}
 
+	// Don't actually create a namespace if the request is for a dry-run.
```

Comment:
I think this can be the first check because the results here won't matter.  We only check the cache to produce this.